src/frontend/src/pages/HomePage/components/OverdueWorkPackages.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { AuthenticatedUser, isAdmin, Team, WorkPackage } from 'shared';
+import { Box, Card, CardContent, Stack, Typography, useTheme } from '@mui/material';
+import WorkPackageCard from './WorkPackageCard';
+import { useAllWorkPackages, useGetManyWorkPackages } from '../../../hooks/work-packages.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import ErrorPage from '../../ErrorPage';
+import { daysOverdue } from '../../../utils/datetime.utils';
+import { PAGE_GRID_HEIGHT } from '../../../components/PageLayout';
+import EmptyPageBlockDisplay from './EmptyPageBlockDisplay';
+import CheckCircleOutlineOutlinedIcon from '@mui/icons-material/CheckCircleOutlineOutlined';
+
+interface OverdueWorkPackagesViewProps {
+  workPackages: WorkPackage[];
+}
+
+interface OverdueWorkPackagesProps {
+  user: AuthenticatedUser;
+}
+
+const NoOverdueWPsDisplay: React.FC = () => {
+  return (
+    <EmptyPageBlockDisplay
+      icon={<CheckCircleOutlineOutlinedIcon sx={{ fontSize: 128 }} />}
+      heading={'Great Job Team!'}
+      message={'Your team has no overdue work packages!'}
+    />
+  );
+};
+
+const getAllWbsNumFromTeams = (teams: Team[]) => {
+  const projects = teams.map((team) => team.projects).flat();
+  const workPackages = projects.map((project) => project.workPackages).flat();
+  return workPackages.map((wp) => wp.wbsNum);
+};
+
+const OverdueWorkPackagesView: React.FC<OverdueWorkPackagesViewProps> = ({ workPackages }) => {
```

Comment:
can we split this file into at least two with the details view and then the container, also the card section could be its own component, I know we already have a work package card component this could go somewhere near that named like OverdueWorkPackageCard

Reply:
yes